plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -400,6 +431,61 @@ protected void copyTableSchema(ConnectorSession session, Connection connection,
         }
     }
 
+    @Override
+    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        Map<String, CaseSensitivity> caseSensitivityMapping = getCaseSensitivityForColumns(session, tableHandle);
+        List<JdbcColumnHandle> columnHandles = super.getColumns(session, tableHandle);
+        if (caseSensitivityMapping.size() > 0) {
+            return columnHandles.stream().map(jdbcColumnHandle -> {
+                JdbcTypeHandle jdbcTypeHandle = jdbcColumnHandle.getJdbcTypeHandle();
+                JdbcTypeHandle enrichedJdbcTypeHandle = new JdbcTypeHandle(
+                        jdbcTypeHandle.getJdbcType(),
+                        jdbcTypeHandle.getJdbcTypeName(),
+                        jdbcTypeHandle.getColumnSize(),
+                        jdbcTypeHandle.getDecimalDigits(),
+                        jdbcTypeHandle.getArrayDimensions(),
+                        Optional.ofNullable(caseSensitivityMapping.get(jdbcColumnHandle.getColumnName())));
+
+                return JdbcColumnHandle.builder()
+                        .setColumnName(jdbcColumnHandle.getColumnName())
+                        .setColumnType(jdbcColumnHandle.getColumnType())
+                        .setJdbcTypeHandle(enrichedJdbcTypeHandle)
+                        .setNullable(jdbcColumnHandle.isNullable())
+                        .setComment(jdbcColumnHandle.getComment())
+                        .build();
+            }).collect(Collectors.toList());
+        }
+
+        return columnHandles;
+    }
+
+    private Map<String, CaseSensitivity> getCaseSensitivityForColumns(ConnectorSession session, JdbcTableHandle tableHandle)
```

Comment:
An extension can be introduced in BaseJdbcClient#getColumns.

Populate a field called `columnCaseSensitivities = getCaseSensitivityForColumns(connection, tableHandle);`

The default impl of that method can return empty map.

Then within BaseJdbcClient#getColumns you use `Optional.ofNullable(columnCaseSensitivities.get(jdbcColumnHandle.getColumnName())`

That way other connectors would benefit as well and we can avoid opening separate connections and also avoid code drift from super class.

Reply:
refactored